### ISSUE TITLE
Access violation when using x64 MSVC C++ Debugger

### ISSUE THREAD

**Issue by esride-jts** [microsoft/vscode-cpptools#4612]
Type: Debugger

**Describe the bug**
- OS and Version: Windows x64 10.0.177763
[main-dump.zip](https://github.com/microsoft/vscode-cpptools/files/3851745/main-dump.zip)

- VS Code Version: 1.40.1
- C/C++ Extension Version: Version 0.26.2-insider: November 12, 2019

**To Reproduce**
*Please include a code sample and `launch.json` configuration.*
1. Compile and debug x64
2. While debugging an "Access Violation" is raised by the nt.dll

**Additional context**
Maybe related to [Access Violation](https://developercommunity.visualstudio.com/content/problem/320458/access-violation-in-combasedll-while-debugging.html)
See attached dump file.

**Comment by pieandcakes** [comment]
Can you tell me which version of the compiler you are using?

**Comment by esride-jts** [comment]
@pieandcakes  the terminal shows Version 19.23.28106.4 for x64

**Comment by esride-jts** [comment]
Any progress on this issue?

**Comment by pieandcakes** [comment]
@esride-jts Sorry I didn't get back to you sooner. I wasn't able to repro this but I have a newer version of the compiler (Compiler Version 19.24.28314). The dump file wasn't very helpful either as it looks like its in the middle of a LoadLibrary call from what I can see.

Are you able to reproduce this behavior with a small Hello World app? The link you posted is related to TrendMicro Antivirus interfering with the debugger. Do you have that or another type of Antivirus installed? Can you try and exclude that folder from Antivirus scanning and seeing if it reproduces?

**Comment by esride-jts** [comment]
@pieandcakes I was able to get the same behaviour using VS 2019 Pro with a simple CMake project. In VS 2019 Pro I was able to resovle it by just enabling the "Automatically close the console when debugging stops" setting. Is this possible in VS Code, too? 

[access violation in combase.dll while debugging](https://developercommunity.visualstudio.com/content/problem/320458/access-violation-in-combasedll-while-debugging.html)

**Comment by pieandcakes** [comment]
@esride-jts If you are using `externalConsole`  set to true, you will need to close it. Otherwise, if its set to false, it shouldn't hold anything open.

**Comment by paedocypris** [comment]
I have the same issue as the original poster, also with a CMake project. However, I can debug the executable generated by calling devenv on it and using visual studio to debug it.

I am using VS2017 if that matters.

**Comment by pieandcakes** [comment]
@paedocypris I am still unable to reproduce it. I've looked at the issue that @esride-jts linked in the original post which points to virus scanning on the executable maybe causing a problem. 

I've tried both a VS 2019 CMake and a VS 2017 CMake C++ hello world project then debugged with this extension and I can't duplicate the read access violation.

**Comment by esride-jts** [comment]
I just moved using to Visual Studio instead. Today I returned back to VS Code and wanted to use the Rust extension. When the Rust extension is using the VS C++ compiler from VS Code the same problem occur. I am not able to turn off this "Trend Micro Office Scan" and the support told me that there must be a first chance access violation in the combase.dll. I would love to use VS Code for native development, but this combase.dll & virus scanning is such a pain in the neck. I am not able to narrow it down any further. Is there a chance to ask some of the VS C++ compiler team if there is a kind of first chance access violation when using the C++ x64 from an external program like VS Code? BTW really great talk regarding "Rust at Microsoft" at the "Microsoft European Virtual Open Source Summit" today.

Thanks in Advance.
